main.ts

```diff
@@ -41,88 +54,114 @@ async function onListen() {
+      return Response.redirect(INVITE_URL);
     }
 
-    case discord.InteractionType.ApplicationCommand: {
-      if (
-        !discord.Utils.isChatInputApplicationCommandInteraction(interaction)
-      ) {
-        return new Response("Invalid request", { status: 400 });
-      }
+    // Verify the request.
+    const { error, body } = await verify(request, env.DISCORD_PUBLIC_KEY);
+    if (error !== null) {
+      return error;
+    }
 
-      if (!interaction.member?.user) {
-        return new Response("Invalid request", { status: 400 });
+    // Parse the incoming request as JSON.
+    const interaction = await JSON.parse(body) as discord.APIInteraction;
+    switch (interaction.type) {
+      case discord.InteractionType.Ping: {
+        return Response.json({ type: discord.InteractionResponseType.Pong });
       }
 
-      if (
-        !interaction.member.roles.some((role) => env.DISCORD_ROLE_ID === role)
-      ) {
-        return new Response("Invalid request", { status: 400 });
-      }
-
-      // Make the Shorter options.
-      const options = makeShorterOptions(
-        interaction.member,
-        interaction.data,
-      );
-
-      // Invoke the Shorter operation.
-      shorter(options)
-        .then((result) =>
-          api.editOriginalInteractionResponse({
-            botID: env.DISCORD_CLIENT_ID,
-            botToken: env.DISCORD_TOKEN,
-            interactionToken: interaction.token,
-            content:
-              `Created commit [${result.message}](https://acmcsuf.com/code/commit/${result.sha})!`,
-          })
-        )
-        .catch((error) => {
-          if (error instanceof Error) {
-            api.editOriginalInteractionResponse({
+      case discord.InteractionType.ApplicationCommand: {
+        if (
+          !discord.Utils.isChatInputApplicationCommandInteraction(interaction)
+        ) {
+          return new Response("Invalid request", { status: 400 });
+        }
+
+        if (!interaction.member?.user) {
+          return new Response("Invalid request", { status: 400 });
+        }
+
+        if (
+          !interaction.member.roles.some((role) => env.DISCORD_ROLE_ID === role)
+        ) {
+          return new Response("Invalid request", { status: 400 });
+        }
+
+        // Make the Shorter options.
+        const options = makeShorterOptions(
+          interaction.member,
+          interaction.data,
+        );
+
+        // Invoke the Shorter operation.
+        shorter(options)
+          .then(async (result) => {
+            // Send the success message.
+            await api.editOriginalInteractionResponse({
               botID: env.DISCORD_CLIENT_ID,
               botToken: env.DISCORD_TOKEN,
               interactionToken: interaction.token,
-              content: `Error: ${error.message}`,
+              content:
+                `Created commit [${result.message}](https://acmcsuf.com/code/commit/${result.sha})!`,
```

Comment:
Add some additional text if a TTL is present. Suggested by @karnikaavelumani.

Reply:
Resolved in <https://github.com/acmcsufoss/shorter/pull/5/commits/ae225af6f39fad4cb991f9f47932b24adb89222b>.